### ISSUE TITLE
Message timestamps are shown in 12h format

### ISSUE THREAD

**Issue by jayb-g** [mollyim/mollyim-android#381]
Is there an existing request for this?
   - [x] I have searched the existing issues

Bug description

All timestamps on messages are 12h format regardless of if device time format is set to 24h. Apparently Signal doesn't have this problem. Maybe molly missed some commits ? Is it intentional?

**Comment by valldrac** [comment]
I tested Signal v7.20.1, it displays time in 12h format when the language is set to English, regardless of device settings.

In Signal, overriding the language sets the app’s locale, which includes the time format. There’s no difference between Molly and Signal on that.

There might be some inconsistencies in certain fields due to a reported bug in Signal where the locale can sometimes reset. However, I haven't been able to replicate this issue, and it's different from what you're mentioning, which is how the language setting is designed to work in the app.

**Comment by valldrac** [comment]
The "system default" option keeps the locale unchanged, meaning the system's time settings stay as they are. But if you switch it to English, the 12h American format will be used.

Have you noticed any differences between Molly and Signal?

**Comment by valldrac** [comment]
> Yes, same settings. Same device. Molly has 12h format. Molly 7.17.5-1.up1-FOSS

Do you have passphrase database encryption enabled in Molly? I found a possible cause related to this.

**Comment by valldrac** [comment]
I was unable to reproduce the issue. When the system language is set to English and 24-hour format, and Molly is set to "system default" the timestamps are correctly displayed in 24-hour format. Restoring a backup or using UnifiedPush shouldn't affect this. The only related bug I found (already reported to Signal) is that after changing the language settings in Android, it's necessary to exit and enter a chat for the changes to take effect.

It’s clear there is likely a bug, but without a reproducible case, it’s difficult to pinpoint the cause. Could you please send the debug log to support@molly.im so I can check the system version and device details? This might help in identifying the problem.

**Comment by valldrac** [comment]
Yes, we was able to reproduce it, but we don't have the fix yet. I'll try to prioritize it.

**Comment by valldrac** [comment]
It's closed as "completed" so it's fixed. See above. The fix will be in the next release.

**Comment by jayb-g** [comment]
Just noticed it now in latest version. It works as expected. Thanks for fixing it 👍
